### PR TITLE
[2.0] Simplify TestAction's API

### PR DESCRIPTION
**Pull request by pepibumur**
Resolves https://github.com/tuist/tuist/issues/YYY
Request for comments document (if applies):

### Short description 📝

As suggested [here](https://github.com/tuist/tuist/issues/2981) by @kwridan, I'm simplifying the `TestActions` methods to create an instance. I removed the methods that take variadic parameters in favor of arrays, and removed the methods that take the configuration as a `String` to take a `PresetBuildConfiguration` instead.

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.

### CODE REVIEW

**Review by danyf90 — APPROVED** [review]

**Review by luispadron — APPROVED** [review]
Minor comments, love seeing all the red ✂️

File: Sources/ProjectDescription/TestAction.swift [hunk 1]
```diff
@@ -65,144 +65,49 @@ public struct TestAction: Equatable, Codable {
...
+            codeCoverageTargets: [],
+            expandVariableFromTarget: nil,
             preActions: preActions,
             postActions: postActions,
-            diagnosticsOptions: diagnosticsOptions,
-            language: language.flatMap(SchemeLanguage.init(stringLiteral:)),
-            region: region
+            diagnosticsOptions: [.mainThreadChecker],
+            language: nil,
+            region: nil
```
Comment (kwridan) [inline Sources/ProjectDescription/TestAction.swift:91]: we'll need to offer an API that allows users to configure/specify some of those parameters that are applicable when using the targets style test action.

**Review by kwridan — COMMENTED** [review]
Thanks for tackling this @pepibumur 

Seems like possibly a bad merge took place, I see the cocoapods changes in this PR which I believe were in a different PR - https://github.com/tuist/tuist/pull/3369

File: Sources/ProjectDescription/TargetDependency.swift [hunk 2]
```diff
@@ -107,8 +101,6 @@ public enum TargetDependency: Codable, Equatable {
             return "package"
         case .sdk:
             return "sdk"
-        case .cocoapods:
-            return "cocoapods"
```
[the remark this answers is not in the document]
Comment (danyf90) [reply Sources/ProjectDescription/TargetDependency.swift]: @kwridan @pepibumur the base branch was main, changed the PR to point to release-2.0

File: Sources/ProjectDescription/TestAction.swift (hunk 1 above)
Comment (pepibumur) [reply Sources/ProjectDescription/TestAction.swift:91]: My bad. I extended the static method to allow passing all the options.

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,10 @@ Please, check out guidelines: https://keepachangelog.com/en/1.0.0/
...
 - **Breaking** Support for deprecated `TuistConfig.swift` has been ended. Define your configuration using `Config.swift`. Check [documentation](https://docs.tuist.io/manifests/config) for details. [#3373](https://github.com/tuist/tuist/pull/3373) by [@laxmorek](https://github.com/laxmorek)
 - **Breaking** Support for deprecated `Template.swift` has been ended. Define your templates using any name that describes them (`name_of_template.swift`). Check [documentation](https://docs.tuist.io/commands/scaffold) for details. [#3373](https://github.com/tuist/tuist/pull/3373) by [@laxmorek](https://github.com/laxmorek)
+  - **Migration:** we recommend wrapping the the generation of projects in a script that runs `pod install` right after generating the project: `tuist generate && pod install`. Alternatively, you might consider adopting Swift Package Manager and using our built-in support for package dependencies through the `Dependencies.swift` manifest file.
```
Comment (fortmarek) [inline CHANGELOG.md:14]: Should we also include the migration steps on the website? That might be a better place than `CHANGELOG`.

**Review by fortmarek — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (pepibumur) [reply CHANGELOG.md]: I think we can do that as part of releasing 2.0 in a new `/migrations` page in the documentation website.

File: Sources/ProjectDescription/TestAction.swift [hunk 4]
```diff
@@ -65,152 +65,87 @@ public struct TestAction: Equatable, Codable {
...
+    ///   - language: The language to be used.
+    ///   - region: The region to be used.
+    /// - Returns: An initialized test action.
     public static func testPlans(_ testPlans: [Path],
-                                 config: PresetBuildConfiguration = .debug,
+                                 arguments: Arguments? = nil,
+                                 configuration: PresetBuildConfiguration = .debug,
+                                 coverage: Bool = false,
+                                 codeCoverageTargets: [TargetReference] = [],
+                                 expandVariableFromTarget: TargetReference? = nil,
```
Comment (kwridan) [inline Sources/ProjectDescription/TestAction.swift:129]: The previous implementation was correct, when using test plans only a subset of those options is available :(

There are two slightly different modes the scheme editor has

When using targets a lot of the options are exposed in the editor directly:

targets



When using test plans however, it hides all the other options as they are defined within the individual test plans:

testplans

**Review by kwridan — COMMENTED** [review]
Thanks for the updates @pepibumur 

Happy to push some of those fixes if that helps?

**Pull request merged**